src/substrait/builders/extended_expression.py

```diff
@@ -205,12 +205,12 @@ def resolve(
 
 
 def scalar_function(
-    urn: str,
-    function: str,
+    function: Union[str, Iterable[str]],
```

Comment:
@tokoko I am open minded to this approach, but I would _strongly_ prefer not introducing string parsing if it is possible. Instead, maybe the API could be something like:
```suggestion
    function: Union[ExtensionID, Iterable[ExtensionID]],
```

where 
```python
@dataclass
class ExtensionID:
  urn: str
  function: str
```

What do you think?

Reply:
I would also be open minded to just having a separate function for the individual and the list case.

Reply:
I'm not a fan of parsing strings either. I simply wanted not to clutter the api too much. How about using `NamedTuple` instead of a dataclass. It would allow the user to pass plain tuples as well.

```
from typing import NamedTuple, Union, Iterable

class ExtensionFunction(NamedTuple):
    urn: str
    function: str

def process_func(func: Union[ExtensionFunction, Iterable[ExtensionFunction]]):
    functions = [func] if isinstance(func[0], str) else func
    for f in functions:
        urn, name = f
        print(urn)
        print(name)

process_func(ExtensionFunction("sample_urn", "sample_func"))
process_func(("sample_urn", "sample_func"))
process_func([("sample_urn", "sample_func1"), ("sample_urn", "sample_func2")])
```

Reply:
I am okay with that approach as an API for allowing multiple parameters, though I still have my hesitations about the PR as a whole as expressed in the below comment.